scripts/build/Platform/Linux/build_linux.sh

```diff
@@ -43,7 +43,26 @@ if [[ ! -z "$RUN_CONFIGURE" ]]; then
     echo "${CONFIGURE_CMD}" > ${LAST_CONFIGURE_CMD_FILE}
 fi
 
-eval echo [ci_build] cmake --build . --target ${CMAKE_TARGET} --config ${CONFIGURATION} -j $(grep -c processor /proc/cpuinfo) -- ${CMAKE_NATIVE_BUILD_ARGS}
-eval cmake --build . --target ${CMAKE_TARGET} --config ${CONFIGURATION} -j $(grep -c processor /proc/cpuinfo) -- ${CMAKE_NATIVE_BUILD_ARGS}
+# In cases where the ratio of cores to memory may cause out of memory issues if we have too many cores,
+# make sure that we have at least 2 GB (2097152 kb) of ram per core and not over allocate the number of jobs
+TOTAL_MEMORY=$(cat /proc/meminfo | grep MemTotal | awk '{print $2}')
+MIN_MEMORY_PER_CORE=2097152
+TOTAL_CORE_COUNT=$(grep -c processor /proc/cpuinfo)
+MAX_CORE_USAGE=$(expr $TOTAL_MEMORY / $MIN_MEMORY_PER_CORE)
+
+echo "Total Memory       : $TOTAL_MEMORY"
+echo "Min Memory Per Job : $MIN_MEMORY_PER_CORE"
+echo "Total Cores        : $TOTAL_CORE_COUNT"
+echo "Max Usable Cores   : $MAX_CORE_USAGE"
+
+if [ $TOTAL_CORE_COUNT -gt $MAX_CORE_USAGE ]
+then
+    CORE_COUNT=$MAX_CORE_USAGE
+else
+    CORE_COUNT=$TOTAL_CORE_COUNT
+fi
```

Comment:
this will make linux jobs slower
The memory limit was passed to Ninja to limit linking, not compilation. This limitation here will limit compilation and linking.
DId you hit memory issues in the GCC job even with 4 linking jobs?

Reply:
I checked on the jenkins nodes, it made it marginally slower. (it used 34 cores instead of 36
```
Total Memory       : 71919544
[2022-01-13T20:04:54.624Z] Min Memory Per Job : 2097152
[2022-01-13T20:04:54.624Z] Total Cores        : 36
[2022-01-13T20:04:54.624Z] Max Usable Cores   : 34
[2022-01-13T20:04:54.624Z] [ci_build] cmake --build . --target all --config profile -j 34 --
```
https://jenkins-pipeline.agscollab.com/blue/organizations/jenkins/O3DE-LY-FORK/detail/spham%2Flbc%2Fgcc_support/8/pipeline/171 

This was crashing gcc compilations (not even linking). With unity turned on, gcc tasks were getting up to 3.5 GB.

Reply:
how much slower? my concern is that linux is currently the critical path (slowest build in AR)
We are hitting 2hs in some cases, so if this adds 10min, it would be a concern
We can use a variable to define this and just do it for these jobs

Reply:
Yeah, i can introduce an environment variable in which I can set it locally so I can run the script without issue